evil-common.el

```diff
@@ -864,9 +864,37 @@ Inhibits echo area messages, mode line updates and cursor changes."
   `(let ((evil-no-display t))
      ,@body))
 
-(defun evil-num-visible-lines ()
-  "Returns the number of currently visible lines."
-  (- (window-height) 1))
+(defvar evil-cached-header-line-height nil
+  "Cached height of the header line.")
+
+(defun evil-header-line-height ()
+  "Return the height of the header line.
+If there is no header line, return nil."
+  (let ((posn (posn-at-x-y 0 0)))
+    (when (eq (posn-area posn) 'header-line)
+      (cdr (posn-object-width-height posn)))))
+
+(defun evil-posn-x-y (position)
+  "Return the x and y coordinates in POSITION.
+
+This function returns y offset from the top of the buffer area including
+the header line.  This definition could be changed in future.
+Note: On Emacs 22 and 23, y offset, returned by `posn-at-point' and taken
+by `posn-at-x-y', is relative to the top of the buffer area including
+the header line.
+However, on Emacs 24, y offset returned by `posn-at-point' is relative to
+the text area excluding the header line, while y offset taken by
+`posn-at-x-y' is relative to the buffer area including the header line.
+This asymmetry is by design according to GNU Emacs team.
+This function fixes the asymmetry between them on Emacs 24 and later versions.
+Borrowed from mozc.el."
+  (let ((xy (posn-x-y position)))
+    (when (and (> emacs-major-version 24) header-line-format)
```

Comment:
Why do do we need this `emacs-major-version` check if the minimum supported version of Emacs is 24.1?

https://github.com/emacs-evil/evil/blob/297b8f323754d967beeaed28080311f5afbe25c8/evil-pkg.el#L5

Reply:
You are right. I learned it from
https://github.com/emacs-evil/evil/blob/297b8f323754d967beeaed28080311f5afbe25c8/evil-commands.el#L3749-L3751
but support for Emacs < 24 was dropped in 2cad47110dd4d1e47fe5691c11566e4fcd2f658d.

I have removed these code.